scheduler/src/lib.rs

```diff
@@ -238,16 +242,18 @@ impl<T: Config> Pallet<T> {
 
 		let tnext = if tnext < now {
 			let gap = now - tnext;
-			let n = gap
-				.checked_div(&cycle_duration)
-				.expect("invalid phase durations: may not be zero");
-			tnext.saturating_add((cycle_duration).saturating_mul(n + T::Moment::one()))
+			if let Some(n) = gap.checked_div(&cycle_duration) {
```

Comment:
The `if let Some(...) else return Err`. Is a more verbose way of writing:

`operation.ok_or_else(|| Err...)?;`